### PR TITLE
Fix issue with Ansible 2.5 for include_tasks

### PR DESCRIPTION
**Pull request by guangie88**
Tested with the following example commands on both Ansible 2.4.3 and 2.5. Also assumes that there is a `secrets.yml` Ansible vault file first:

```bash
ansible-playbook -i inventory vault.yml -t generate_key
ansible-playbook -i inventory --ask-vault-pass vault.yml -t encrypt
ansible-playbook -i inventory vault.yml -t decrypt
```

### CODE REVIEW

**Comment by guangie88** [comment]
Meant to resolve issue #4.

File: roles/aes-decrypt/tasks/main.yml [hunk 1]
```diff
@@ -1,23 +1,26 @@
...
-- name: Decrypt data
-  shell: |
-    openssl enc \
-      -aes-256-cbc \
-      -d \
-      -in "{{ ciphertext_temp.path }}" \
-      -out "{{ plaintext }}" \
-      -K "{{ key_hex }}" \
-      -iv "{{ iv.stdout }}"
+- block:
```
Comment (lawliet89) [inline roles/aes-decrypt/tasks/main.yml:2]: Do roles need to have the tags?

Or the same behaviour also applies to `include_role`?

File: roles/aes-decrypt/tasks/main.yml [hunk 2]
```diff
@@ -1,23 +1,26 @@
...
+    shell: |
+      openssl enc \
+        -aes-256-cbc \
+        -d \
+        -in "{{ ciphertext_temp.path }}" \
+        -out "{{ plaintext }}" \
+        -K "{{ key_hex }}" \
+        -iv "{{ iv.stdout }}"
+  tags:
+    - decrypt
```
Comment (lawliet89) [inline roles/aes-decrypt/tasks/main.yml]: Please including a trailing new line in all of your files.

Might want to configure your text editor accordingly.

File: tasks/read_key.yml [hunk 3]
```diff
@@ -2,29 +2,33 @@
...
+          fact_key: "key"
+          payload: "{{ key_json.CiphertextBlob }}"
+    when: key_json.Plaintext is not defined
+  ##############################
+  # Convert key (plaintext) to hex
+  ##############################
+  - name: Convert key (plaintext) to hex
+    set_fact: {"{{ fact_key | default('key_hex') }}":"{{ key | base64_to_hex }}"}
+  tags:
+    - encrypt
```
Comment (lawliet89) [inline tasks/read_key.yml]: Should just be `always`.

**Review by lawliet89 — CHANGES_REQUESTED** [review]
- IIRC, I don't think roles need the tags. 
- Please have a trailing new line for your files.

File: roles/aes-decrypt/tasks/main.yml (hunk 1 above)
Comment (guangie88) [reply roles/aes-decrypt/tasks/main.yml:2]: I tried to revert all `tasks/main.yml` in roles, it doesn't work for 2.5. I guess the files here are considered just like tasks.

File: roles/aes-decrypt/tasks/main.yml (hunk 2 above)
Comment (guangie88) [reply roles/aes-decrypt/tasks/main.yml]: Okay fixed, and found the option to enable the trailing new line on save haha 😛

File: tasks/read_key.yml (hunk 3 above)
Comment (guangie88) [reply tasks/read_key.yml]: Okay fixed.

**Comment by guangie88** [comment]
Thanks for feedback, fixed the issues and re-tested against `ansible` 2.4.3 and 2.5 and working fine.

File: roles/aes-decrypt/tasks/main.yml (hunk 1 above)
Comment (lawliet89) [reply roles/aes-decrypt/tasks/main.yml:2]: So `import|include_role` simply inlines the tasks as you observed. See https://docs.ansible.com/ansible/devel/user_guide/playbooks_reuse.html

Wonder if we can get around this with static imports instead. 

Why must Ansible developers make the behaviour so strange and complicated? >_>

**Review by lawliet89 — APPROVED** [review]

**Comment by guangie88** [comment]
It is a pretty breaking change, and not really for the better :disappointed: The tendency is to make people add `tags: always` for all included tasks/role's tasks.

**Pull request merged**